components/org.wso2.micro.integrator.initializer/src/main/java/org/wso2/micro/integrator/initializer/deployment/AppDeployerServiceComponent.java

```diff
@@ -75,13 +82,19 @@ protected void activate(ComponentContext ctxt) {
         // Deploy artifacts
         artifactDeploymentManager.deploy();
 
-        // Deploy carbon applications
-        try {
-            cAppDeploymentManager.deploy();
-        } catch (CarbonException e) {
-            log.error("Error occurred while deploying carbon application", e);
+        Parameter hotDeployment = configCtx.getAxisConfiguration().getParameter(TAG_HOT_DEPLOYMENT);
+        if (hotDeployment != null && JavaUtils.isTrue(hotDeployment.getValue(), true)) {
+            // Invoke registered deployers, so that they will be triggered followed by hot deployment.
+            invokeRegisteredDeployers();
+        } else {
+            try {
+                // Deploy all carbon applications in cApp directory during the startup time.
+                cAppDeploymentManager.deploy();
```

Comment:
Regardless of whether hot deployment is enabled we deploy synapse artifacts (I suppose?) at line 83, then we only deviate the path of CAPP deployment. Is this intentional?

Reply:
This is now corrected to have only one method called invokeRegisteredDeployers(), hence one path to handle the artifact(.dbs, .xml) and capp deployment for both hot deployment true and false cases.

When the hot deployment is enabled, CarbonDeploymentSchedulerTask will be scheduled through DeploymentEngine class to handle hot deployment and will not be otherwise.